shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs

```diff
@@ -1458,3 +1460,60 @@ fn test_cassandra_peers_rewrite() {
         );
     }
 }
+
+#[tokio::test(flavor = "multi_thread")]
+#[serial]
+async fn test_cassandra_request_throttling() {
+    let _docker_compose = DockerCompose::new("examples/cassandra-passthrough/docker-compose.yml")
+        .wait_for_n_t("Startup complete", 1, 90);
+
+    let shotover_manager = ShotoverManager::from_topology_file(
+        "tests/test-topologies/cassandra-request-throttling.yaml",
+    );
+
+    let connection = shotover_manager.cassandra_connection("127.0.0.1", 9042);
+    std::thread::sleep(std::time::Duration::from_secs(1));
+    let connection_2 = shotover_manager.cassandra_connection("127.0.0.1", 9042);
+
+    // sleep to reset the window
+    std::thread::sleep(std::time::Duration::from_secs(1));
+
+    let statement = stmt!("SELECT * FROM system.peers");
+
+    let mut futures = vec![];
+    // these should all be let through the request throttling
+    for _ in 0..10 {
+        futures.push(connection.execute(&statement));
+        futures.push(connection_2.execute(&statement));
+    }
+    try_join_all(futures).await.unwrap();
+
+    // sleep to reset the window
+    std::thread::sleep(std::time::Duration::from_secs(1));
+
+    futures = vec![];
+    // only around half of these should be let through the request throttling
+    for _ in 0..20 {
+        futures.push(connection.execute(&statement));
+        futures.push(connection_2.execute(&statement));
+    }
+    let mut results = join_all(futures).await;
+    results.retain(|result| match result {
+        Ok(_) => true,
+        Err(e) => {
+            if let Error(
+                ErrorKind::CassErrorResult(cassandra_cpp::CassErrorCode::SERVER_OVERLOADED, ..),
+                _,
+            ) = e
+            {
+            } else {
+                panic!(
+                    "wrong error returned, got {:?}, expected SERVER_OVERLOADED",
+                    e
+                )
+            }
+            false
+        }
+    });
+    assert!(results.len() < 22 && results.len() > 18);
```

Comment:
how did you arrive at the values of 22 and 18?

Reply:
Running it a few times and seeing the variance.

Reply:
If it reached 22 and 18 maybe we should bump it to 25 and 15?

Also where are these inaccuracies in the rate limiter coming from?

Reply:
From Cassandra not taking the exact same amount of time to respond to each request.

Reply:
I can understand needing 20 < results.len() < 22

But I dont understand why we need 18 < results.len() < 22
If the limit is 20 then I dont see any way to time the messages such that only 18 get through.
So either:
* the test should be changed to 20 < results.len() < 22
* there is a bug in the governor crate or we dont understand how the governor crate works